### PR TITLE
fix(predict): correct `class_name` for pretrained COCO models

### LINKED ISSUE #988: v1.6.4 Wrong predicted class names
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

```py
from rfdetr import RFDETRSegSmall

model = RFDETRSegSmall()

detections = model.predict("https://media.roboflow.com/dog.jpg", threshold=0.5)
print(detections.data["class_name"])
```
It output: ['sheep' 'tie' 'motorcycle']. However, it supposed to output "dog".

### Environment

RF-DETR 1.6.4

### Minimal Reproducible Example

```py
from rfdetr import RFDETRSegSmall

model = RFDETRSegSmall()

detections = model.predict("https://media.roboflow.com/dog.jpg", threshold=0.5)
print(detections.data["class_name"])
```

### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Borda**
This pull request fixes and improves how COCO class IDs are mapped to class names in predictions, especially for pretrained COCO models that use sparse category IDs (with gaps) rather than contiguous indices. It ensures correct class name assignment for both pretrained and fine-tuned models, and adds comprehensive tests to cover these scenarios and prevent regressions.

**COCO class ID mapping improvements:**

* Added logic in `predict` (in `rfdetr/detr.py`) to detect when a model is a pretrained COCO checkpoint (using sparse COCO category IDs) and properly map those IDs to class names, rather than assuming a direct index. This fixes incorrect class name assignments for COCO-pretrained models. [[1]](diffhunk://#diff-88cb2cb1c972ea4e1b9e1582f78beb4ad32bf7c10709dd837c57a2da1ce98c86R1397-R1415) [[2]](diffhunk://#diff-88cb2cb1c972ea4e1b9e1582f78beb4ad32bf7c10709dd837c57a2da1ce98c86L1427-R1464)
* Introduced `_SORTED_COCO_IDS` at import time for efficient mapping from COCO category IDs to class names. [[1]](diffhunk://#diff-88cb2cb1c972ea4e1b9e1582f78beb4ad32bf7c10709dd837c57a2da1ce98c86L33-R33) [[2]](diffhunk://#diff-88cb2cb1c972ea4e1b9e1582f78beb4ad32bf7c10709dd837c57a2da1ce98c86R53-R55)

**Testing and regression coverage:**

* Added multiple parameterized and scenario-specific tests to ensure correct class name mapping for:
  - Pretrained COCO models with sparse IDs,
  - Pretrained models with different `dataset_file` values,
  - Fine-tuned models with contiguous IDs,
  - Custom class names with high `num_classes`.
  These tests prevent regressions and cover edge cases reported in previous issues.

---

- Drop `_dataset_file in ("coco", None)` guard from `_is_coco_pretrained`: pretrained COCO checkpoints packaged via Roboflow have `dataset_file="roboflow"`, causing PR #1005 fix to never fire
- Two conditions now sufficient: `num_logit_slots > n` AND `class_names == COCO_CLASS_NAMES`
- Build sparse `_class_id_to_name` dict pairing sorted COCO IDs with class_names; raw ID 18 → "dog" instead of `class_names[18]` = "sheep"
- Add `_SORTED_COCO_IDS` module-level constant to avoid re-sorting on every predict() call
- Add 4 regression tests: sparse mapping (parametrized 3/18/27), dataset_file="roboflow" guard, fine-tuned direct-indexing guard, custom-names guard

resolves #988

### CODE REVIEW

File: src/rfdetr/detr.py [hunk 1]
```diff
@@ -1391,6 +1394,25 @@ def predict(
...
+        if _model_args is None and model_class_names == list(COCO_CLASS_NAMES):
+            logger.warning_once(
+                "predict(): model has no 'args' attribute — COCO sparse-ID mapping cannot activate; "
+                "class_ids are treated as 0-indexed (may be wrong for pretrained COCO checkpoints)"
+            )
+        num_logit_slots: int = getattr(_model_args, "num_classes", n) if _model_args is not None else n
+        _is_coco_pretrained = num_logit_slots > n and model_class_names == list(COCO_CLASS_NAMES)
+        if _is_coco_pretrained:
+            _class_id_to_name: dict[int, str] = {
+                coco_id: model_class_names[i] for i, coco_id in enumerate(_SORTED_COCO_IDS) if i < n
```
Comment (Borda) [inline src/rfdetr/detr.py]:
```suggestion
                coco_id: model_class_names[i] for i, coco_id in enumerate(COCO_CLASSES) if i < n
```

File: src/rfdetr/detr.py [hunk 2]
```diff
@@ -1391,6 +1394,25 @@ def predict(
...
+        # args.num_classes > len(class_names) AND class_names == COCO_CLASS_NAMES.
+        # Fine-tuned models remap category IDs to 0-based contiguous indices, so
+        # class_id i maps directly to class_names[i].
+        _model_args = getattr(self.model, "args", None)
+        if _model_args is None and model_class_names == list(COCO_CLASS_NAMES):
+            logger.warning_once(
+                "predict(): model has no 'args' attribute — COCO sparse-ID mapping cannot activate; "
+                "class_ids are treated as 0-indexed (may be wrong for pretrained COCO checkpoints)"
+            )
+        num_logit_slots: int = getattr(_model_args, "num_classes", n) if _model_args is not None else n
```
Comment (Borda) [inline src/rfdetr/detr.py]:
```suggestion
        num_logit_slots: int = getattr(_model_args, "num_classes", n)
```
